nifi-nar-bundles/nifi-confluent-platform-bundle/nifi-confluent-schema-registry-service/src/main/java/org/apache/nifi/confluent/schemaregistry/ConfluentSchemaRegistry.java

```diff
@@ -112,6 +114,32 @@
         .required(true)
         .build();
 
+    static final PropertyDescriptor PROP_BASIC_AUTH_USERNAME = new PropertyDescriptor.Builder()
+        .name("Authentication Username")
+        .displayName("Authentication Username")
+        .description("The username to be used by the client to authenticate against the Remote URL.  Cannot include control characters (0-31), ':', or DEL (127).")
+        .required(false)
+        .addValidator(StandardValidators.createRegexMatchingValidator(Pattern.compile("^[\\x20-\\x39\\x3b-\\x7e\\x80-\\xff]+$")))
+        .build();
+
+    static final PropertyDescriptor PROP_BASIC_AUTH_PASSWORD = new PropertyDescriptor.Builder()
+        .name("Authentication Password")
+        .displayName("Authentication Password")
+        .description("The password to be used by the client to authenticate against the Remote URL.")
+        .required(false)
+        .sensitive(true)
+        .addValidator(StandardValidators.createRegexMatchingValidator(Pattern.compile("^[\\x20-\\x7e\\x80-\\xff]+$")))
+        .build();
+
+    static final PropertyDescriptor PROP_AUTH_TYPE = new PropertyDescriptor.Builder()
+        .name("Authentication Type")
+        .displayName("Authentication Type")
+        .description("Basic authentication will use the 'Authentication Username' " +
+                "and 'Authentication Password' property values. See Confluent Schema Registry documentation for more details.")
+        .required(false)
+        .allowableValues("BASIC")
```

Comment:
I don't think setting the allowable value here to a static string is sufficient to enforce that the provided value will always be provided in all uppercase (via API call or manipulated submission), and there is no default value to offer an absence of authentication.

Reply:
Your point is well taken about API generated configurations bypassing the UI validation.  All comparisons will this value are now case insensitive.   I don't think we can really set a default value here, as the default is the absence of value (unset) unless .default("") == unset?

Reply:
The [Confluent Server documentation](https://docs.confluent.io/platform/current/kafka-rest/production-deployment/confluent-server/security.html#http-basic-authentication) lists BASIC and NONE as options for the authentication method for certain clients.  Following that approach and having a default value of NONE, in combination with an AuthenticationType enum, would help make the configuration properties and implementation approach clearer.